packages/inference/src/tasks/custom/request.ts

```diff
@@ -16,7 +16,7 @@ export async function request<T>(
 	const { url, info } = await makeRequestOptions(args, options);
 	const response = await (options?.fetch ?? fetch)(url, info);
 
-	if (options?.retry_on_error !== false && response.status === 503 && !options?.wait_for_model) {
+	if (options?.retry_on_error !== false && response.status === 503 && options?.wait_for_model) {
```

Comment:
```suggestion
	if (options?.retry_on_error !== false && response.status === 503 && options?.wait_for_model === undefined) {
```

same for streaming

current behavior is wanted (to retry with `wait_for_model` if we get a 503)